sql-plan: add description about plan baseline (#13318)
pingcap/docs: sql-plan-management.md

Used for [preventing regression of execution plans during an upgrade](#prevent-regression-of-execution-plans-during-an-upgrade), this feature captures queries that meet capturing conditions and creates bindings for these queries.

> A plan baseline refers to a collection of approved plans that the optimizer can use for executing a SQL statement. Generally, a plan is added to the plan baseline by the database after confirming that it delivers satisfactory performance. A plan in this context encompasses all the necessary plan-related details (such as SQL plan identifier, hint set, bind values, and optimizer environment) that the optimizer requires to replicate an execution plan.

Reviewer: ```suggestion
A plan baseline refers to a collection of accepted plans that the optimizer can use for executing a SQL statement. Generally, TiDB adds a plan into the plan baseline only after confirming that the plan delivers satisfactory performance. A plan in this context encompasses all the necessary plan-related details (such as SQL plan identifier, hint set, bind values, and optimizer environment) that the optimizer requires to reproduce an execution plan.
```
